api/envoy/admin/v2alpha/config_dump.proto

```diff
@@ -45,6 +45,22 @@ message BootstrapConfigDump {
   google.protobuf.Timestamp last_updated = 2;
 }
 
+// This is a general purpose proto for dynamic configuration to track information about both
+// currently loaded configuration and the most recent attempt to load configuration.
+message ConfigUpdateState {
+  // The name of the entity whose state is tracked here.
+  string name = 1;
+
+  // The timestamp of the last update attempt.
+  google.protobuf.Timestamp last_update = 2;
+
+  // Simple tracking if the last update attempt was successful.
+  bool update_successful = 3;
+
+  // Details about the last update attempt
+  string last_update_details = 4;
```

Comment:
Based on code, it seems it is only populated on error - probably be explicit about it and call `last_error_details`?